### PR TITLE
[TH-187] Add get pool list/detail api

### PR DESCRIPTION
**Pull request by ssk7808**
# Description

- get pool list, detail api 작성했습니다

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: src/api/pools/pools-get.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      cursor: number;
+      take: number;
+      totalCount: number;
+      totalPage: number;
+      hasNextPage: number;
+    };
+    // sort params를 넣어서 호출한 경우
+    sort: {
+      property: string;
+      direction: string;
```
Comment (Imchaemin) [inline src/api/pools/pools-get.ts:33]:
```suggestion
      property?: string;
      direction?: string;
```

File: src/api/pools/pools-get.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    };
+    // filter params를 넣어서 호출한 경우
+    filter: {
+      property: string;
+      rule: string;
+      value: string;
```
Comment (Imchaemin) [inline src/api/pools/pools-get.ts:39]:
```suggestion
      property?: string;
      rule?: string;
      value?: string;
```

File: src/api/pools/pools-get.ts (hunk 1 above)
Comment (Imchaemin) [reply src/api/pools/pools-get.ts:33]: sort param이 없으면 `sort: {}` 로 내려옵니당

File: src/api/pools/pools-get.ts (hunk 2 above)
Comment (Imchaemin) [reply src/api/pools/pools-get.ts:39]: 마찬가지

File: src/api/pools/pools-get.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  useQuery<PoolsResponse>(['pools', 'get-pools', params], () => getPoolsAxios(params), {
+    ...options,
+  });
+
+interface PoolByIdResponse {
+  pool: IPool;
+}
+
+const getPoolsByIdAxios = async (networkAbbr: string, poolId: string) =>
+  (await api.get<PoolByIdResponse>(`/pool/:${networkAbbr}/:${poolId}`)).data;
```
Comment (Imchaemin) [inline src/api/pools/pools-get.ts:59]:
```suggestion
  (await api.get<PoolByIdResponse>(`/pool/${networkAbbr}/${poolId}`)).data;
```

**Review by Imchaemin — COMMENTED** [review]
response 타입도 조금 달라졌습니다. 확인해주시면 감사하겠습니다
IPool은 노션에 적혀있는것에서 liquidityProvision, swapHistories 만 제거해주시면 됩니다.

File: src/api/pools/pools-get.ts (hunk 3 above)
Comment (ssk7808) [reply src/api/pools/pools-get.ts]: [fix: get pool null handling, fix get api url](https://github.com/TeamHeimdallr/moai-web/pull/165/commits/2f6214cb5e154f76350211a210f7cbeda8d6fcf6)

File: src/api/pools/pools-get.ts (hunk 2 above)
Comment (ssk7808) [reply src/api/pools/pools-get.ts]: [fix: get pool null handling, fix get api url](https://github.com/TeamHeimdallr/moai-web/pull/165/commits/2f6214cb5e154f76350211a210f7cbeda8d6fcf6)

File: src/api/pools/pools-get.ts (hunk 1 above)
Comment (ssk7808) [reply src/api/pools/pools-get.ts]: [fix: get pool null handling, fix get api url](https://github.com/TeamHeimdallr/moai-web/pull/165/commits/2f6214cb5e154f76350211a210f7cbeda8d6fcf6)

**Comment by Imchaemin** [comment]
image

타입수정은 Contract => API로 수정하면서 동시 진행

**Review by Imchaemin — APPROVED** [review]

**Pull request merged**